oss_fuzz_integration/project_checker.py

```diff
@@ -144,12 +145,30 @@ def check_specific_project(proj_name, build_log_file,coverage_log,summary_json):
     print("Check done")
 
 
+def check_project_html_sanity(html_report):
+    """Checks the sanity of URLs in the html file. Returns False if fails"""
+    with open(html_report, 'r') as html_file:
+        for line in html_file:
+            line_pos = line.find("<a href")
+            if line_pos != -1:
+                url_begin_pos = line.find('"', line_pos)
+                url_end_pos = line.find('"', url_begin_pos+1)
+                url = line[url_begin_pos+1:url_end_pos]
+                if url.startswith("http"):
+                    status = requests.head(url)
```

Comment:
Yes, I test these with these [steps](https://github.com/ossf/fuzz-introspector/tree/main/oss_fuzz_integration#testing-before-bumping-oss-fuzz), and it catches links like this:

https://github.com/ossf/fuzz-introspector/tree/main//doc/Glossary.md#call-tree-overview

>  Doesn't this line presume a server is currently running and serving the HTML report?

Yes, it only sends request if the url starts with `http`, so local links are fine.

Reply:
You are right! it is not catching the exact bug that made me to write this test. Because on our local tests, we do not use coverage reports via GCB urls. 

But as it helped to catch another kind of url bugs, I am still inclined to have this merged.

Reply:
And to be more precise, the bug in #362 is not exactly reproducible locally, it only happens on our cloud builds.

Reply:
Okay -- can we update the comment on the PR then with the specific type of URLs it checks for? Because it references https://github.com/ossf/fuzz-introspector/issues/362 so I thought it was a regression check.

I think we can write a unit test that checks for reported issue though, e.g. by having a tainted .yaml file and then coding it up ensure the post processing does not use the tainted URL without correcting it. See this PR https://github.com/ossf/fuzz-introspector/pull/402 for an example where I fix a regression related to URLs and add a unit test that checks the regression.

Reply:
> can we update the comment 

good point! Done.

>coding it up ensure the post processing does not use the tainted URL without correcting it.


Yes that could work, but it is leaving the source of taint still there! I did not want to have a url sanitizer in post-process. The change in #405 should fix the issue from the source.